addons/purchase_stock/wizard/product_replenish.py

```diff
@@ -42,7 +42,7 @@ def _compute_show_vendor(self):
         for rec in self:
             rec.show_vendor = rec._get_show_vendor(rec.route_id)
 
-    @api.onchange('route_id')
+    @api.depends('route_id')
     def _onchange_route_id(self):
```

Comment:
This isn't the correct fix. Warnings can only be returned by onchanges and onchanges must always have an @api.onchange. You will need to add in a check somewhere to prevent the looping.